### PR TITLE
Try to find previous non-empty line for correct indentation

### LINKED ISSUE #1980: Indentation falls back to column 0 after second line break
In latest master, type the following:

```fsharp
let getStringAsync(str: string) = async { return str }

let fooAsync() = async {
    let! str1 = getStringAsync "hello"    
}
```
Now, hit enter.  Notice that the cursor is at the same column as the above `let!` binding,  column 5:

```fsharp
let getStringAsync(str: string) = async { return str }

let fooAsync() = async {
    let! str1 = getStringAsync "hello"    
//  |  <--- cursor is here
}
```

Now, hit enter again:

**Expected:** Cursor is on a new line, and still on column 5.

**Actual:** Cursor is at column 1.

screen shot 2016-12-09 at 11 00 36 am

### PR DESCRIPTION
**Pull request by dungpa**
Fix #1980.

### CODE REVIEW

**Comment by abelbraaksma** [comment]
Does this also fix (in part of whole) #1588?

**Comment by dungpa** [comment]
@abelbraaksma It should fix #1588. 
Looking at the code, I don't think Visual F# Tools has supported Smart Indent, whatever it means.

**Comment by KevinRansom** [comment]
@dungpa 
If you want to force a pr to be re-run through the ci.

Type 
````
@dotnet-bot test this please
````
This will not restart appveyor, but will run all of the other ci scenarios.

Kevin

File: vsintegration/src/FSharp.Editor/IndentationService.fs [hunk 1]
```diff
@@ -30,21 +30,33 @@ open Microsoft.FSharp.Compiler.SourceCodeServices
...
+            match tryFindPreviousNonEmptyLine lineNumber with
+            | None -> Some 0
+            | Some previousLine ->          
+                let rec loop column spaces =
+                    if previousLine.Start + column >= previousLine.End then
+                        spaces
+                    else 
+                        match previousLine.Text.[previousLine.Start + column] with
+                        | ' ' -> loop (column + 1) (spaces + 1)
+                        | '\t' -> loop (column + 1) (((spaces / tabSize) + 1) * tabSize)
```
Comment (KevinRansom) [inline vsintegration/src/FSharp.Editor/IndentationService.fs]: Do we allow \t to appear in an F# source file in the editor?  is it not immediately turned into spaces?

Comment (KevinRansom) [inline vsintegration/src/FSharp.Editor/IndentationService.fs]: Hmm perhaps we do in the not light mode. never mind.

**Comment by dungpa** [comment]
@KevinRansom Thanks for the tip.

**Comment by KevinRansom** [comment]
@dungpa 
Thank you for taking care of this

Kevin

**Pull request merged**